src/Doctrine/ReadModel/CacheDocumentRepository.php

```diff
@@ -35,17 +38,24 @@ public function fetch(string $id, bool $includeMetadata = false): JsonDocument
         return new JsonDocument($id, $value);
     }
 
-    public function save(JsonDocument $document): void
+    public function save(JsonDocument $document, int $attempts = 3): void
     {
         $saved = $this->cache->save($document->getId(), $document->getRawBody(), 0);
 
         if (!$saved) {
-            throw new RuntimeException('Could not save document ' . $document->getId() . 'to cache.');
+            throw new RuntimeException('Could not save document ' . $document->getId() . ' to cache.');
         }
 
         $savedDocument = $this->fetch($document->getId());
         if ($savedDocument->getRawBody() !== $document->getRawBody()) {
-            $this->logger->error('Saved document in cache does not match provided document ' . $document->getId());
+            $this->logger->error(
+                'Saved document in cache does not match provided document ' . $document->getId() . '. Retry attempts left: ' . $attempts
+            );
```

Comment:
`error()` logs are also sent to Sentry. Maybe make this a `debug()` instead (which is only saved to a file), and add a new `error()` for when it is not saved even after the max numer of retries? Only in the latter case do we need to be alerted in Slack I think, so we can replay the event/place/organizer later to fix it.

Reply:
I used log level `warning` when there are attempts left, because the save should never fail.

Reply:
Good choice!